src/components/action-popover/action-popover-menu/action-popover-menu.component.tsx

```diff
@@ -108,46 +116,92 @@ const ActionPopoverMenu = React.forwardRef<
       });
     }, [children]);
 
+    const isItemDisabled = useCallback(
+      (value: number) => {
+        const item = items[value];
+        // The invariant will be triggered before this else path can be explored, hence the ignore else.
+        // istanbul ignore else
+        return React.isValidElement(item) && item.props.disabled;
+      },
+      [items]
+    );
+
+    const firstFocusableItem = items.findIndex(
+      (_, index) => !isItemDisabled(index)
+    );
+
+    const lastFocusableItem =
+      items.length -
+      1 -
+      items
+        .slice()
+        .reverse()
+        .findIndex((_, index) => !isItemDisabled(items.length - 1 - index));
```

Comment:
suggestion: we could try and improve this search a little

If we do a reverse loop we can just find the first non-disabled item which should be a little more readable and more performative by not having to make a copy of the array and reversing it

```suggestion
function getLastFocusableItemAlternative(items, isItemDisabled) {
  let lastFocusableItem = -1;
  for (let i = items.length - 1; i >= 0; i--) {
    if (!isItemDisabled(i)) {
      lastFocusableItem = i;
      break;
    }
  }
  return lastFocusableItem;
}
```